.github/workflows/default.yaml

```diff
@@ -107,7 +114,7 @@ jobs:
           sudo sed --in-place -E 's/# (deb-src.*updates main)/  \1/g' /etc/apt/sources.list
           sudo apt-get --quiet=2 update
           sudo apt-get --quiet=2 build-dep squid
-          sudo apt-get --quiet=2 install linuxdoc-tools libtool-bin
+          sudo apt-get --quiet=2 install linuxdoc-tools libtool-bin clang
```

Comment:
But we may also drop this requirement to install everything every time
and use the pre-build docker images we maintain.
This would free us from the constraint of only using the couple of
Ubuntu versions that github supports, and give us a full alternative to
jenkins

Reply:
> But we may also drop this requirement to install everything every time and use the pre-build docker images we maintain. 

The current approach is much better for the existing simple cases IMO. I can detail why, but this PR is probably not the best place to debate what future unrelated changes are beneficial.

> This would free us from the constraint of only using the couple of Ubuntu versions that github supports, and give us a full alternative to jenkins

Please do not interpret my response above as a vote against using docker images on GitHub Actions. The two approaches are not mutually exclusive (although they may benefit from being located in different Action _jobs_).

Reply:
Let us leave installing a full environment on, for now. We will be able to revise in the future if there is benefit

Reply:
> Let us leave installing a full environment on, for now. We will be able to revise in the future if there is benefit

Your suggestion is compatible with this change request IMO, but we may define "full environment" differently, or perhaps the discussion drifted away too far from the change request itself.

If it is possible to address this change request without heroic efforts,  please do that (to provide the benefits this change request outlines). I see no signs that it is impossible.

FWIW, I would lean towards the second sketch among the two suggested earlier:

```suggestion
          sudo apt-get --quiet=2 install linuxdoc-tools libtool-bin ${{ matrix.compiler.CC }}
```

Reply:
Are you saying you will block this PR unless this change is implemented?
I'm not keen on it, because it makes the build environments more different than strictly needed, so unless it's the only way to land the change, I would prefer to not do it.

Reply:
> Are you saying you will block this PR unless this change is implemented?

No, I am not saying that. I am also not saying that this PR will be approved without changes. As you know, I strive to judge code based on its merits rather than use ultimatums and power plays. If you find errors in my logic, and the fixed logic leads to a different decision, then either no or different PR changes will be needed.

> I'm not keen on it, because it makes the build environments more different than strictly needed

Thank you for disclosing the basis for your proposal to install clang where it is not used. IMO, the priority/weight of that "similar build environments" goal is far lower/smaller than that of "more representative build environments" and "more pure build environments" goals[^G]. The "similarity" goal has little to do with test realism/quality/usefulness -- similarity is primarily a test code clarity and maintenance simplification goal. The suggested change does not complicate test code understanding and maintenance by much (if at all).

[^G]: These short "quoted" summaries may not reflect all the important nuances of each goal characteristic well, but I hope you get the idea.

In the ideal world of unlimited resources, we would have tested all four combinations:

1. Using gcc on Ubuntu with gcc and without clang installed.
2. Using gcc on Ubuntu with gcc and with clang installed.
3. Using clang on Ubuntu without gcc and with clang installed.
4. Using clang on Ubuntu with gcc and with clang installed.

This PR proposes to pick two out of four possible builds. I assume[^A] that removing (or not installing) gcc is difficult, so we cross out the only "without gcc" item -- item 3. After that, we have to use item 4 to get any clang tests. Out of the remaining two builds (item 1 and item 2), the first one is arguably more "representative" (fewer folks install _both_ compilers) and clearly more "pure" (i.e. has fewer unused components) than the second. Thus, item 2 looses, and we are left with the final two:

1: Using gcc on Ubuntu with gcc and without clang installed.
4: Using clang on Ubuntu with gcc and with clang installed.

[^A]: As always, please correct any wrong assumptions. If this assumption is wrong, then we should probably use "pure" items 1 and 3 instead of items 1 and 4.